boa/src/builtins/uri/mod.rs

```diff
@@ -0,0 +1,108 @@
+//! This module implements the global `decodeURI` and encodURI functions.
+
+use std::borrow::Borrow;
+
+use crate::value::RcString;
+use crate::{
+    builtins::BuiltIn, object::FunctionBuilder, property::Attribute, value::Value, BoaProfiler,
+    Context, Result,
+};
+use percent_encoding::{percent_decode, utf8_percent_encode, AsciiSet, CONTROLS};
+
+type EncodeFuncType = fn(&RcString) -> Value;
+
+// https://url.spec.whatwg.org/#fragment-percent-encode-set
+const ENCODE_FRAGMENT: &AsciiSet = &CONTROLS.add(b' ').add(b'"').add(b'<').add(b'>').add(b'`');
+
+fn decode(str: &RcString) -> Value {
```

Comment:
I think we should not call variables `str`, since that is a type in Rust. Better if you call it `arg` or something like that.